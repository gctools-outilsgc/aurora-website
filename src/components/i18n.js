import i18n from "i18next";
import Backend from "i18next-xhr-backend";
import LanguageDetector from "i18next-browser-languagedetector";
import { reactI18nextModule } from "react-i18next";

i18n
  .use(Backend)
  .use(LanguageDetector)
  .use(reactI18nextModule) // if not using I18nextProvider
  .init({
    backend: {      // for all available options read the backend's repository readme file
      loadPath: (!process.env.NODE_ENV || process.env.NODE_ENV === 'development') ?
        '/locales/{{lng}}/{{ns}}.json'
        :
        '/design-system-code/src/locales/{{lng}}/{{ns}}.json'
    }
    ,
    lng: "en", 
    fallbackLng: 'en',
    debug: true,
    interpolation: {
      escapeValue: false, // not needed for react!!
    },

    // react i18next special options (optional)
    react: {
      wait: false,
      bindI18n: 'languageChanged loaded',
      bindStore: 'added removed',
      nsMode: 'default'
    }
  });


export default i18n;