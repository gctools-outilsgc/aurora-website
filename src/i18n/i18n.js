import i18n from "i18next";
import enTranslation from './locales/en/translation';
import frTranslation from './locales/fr/translation';

i18n
  .init({
    lng: "en", 
    fallbackLng: 'en',
    debug: true,
    resources: {
      en: {
        translation: enTranslation
      },
      fr: {
        translation: frTranslation
      }
    },
    interpolation: {
      escapeValue: false, // not needed for react!!
    },

    // react i18next special options (optional)
    react: {
      wait: false,
      bindI18n: 'languageChanged loaded',
      bindStore: 'added removed',
      nsMode: 'default'
    }
  });

export default i18n;