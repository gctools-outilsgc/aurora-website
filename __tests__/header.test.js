import React from "react";
import renderer from "react-test-renderer";
import Header from "../src/components/header";
import { I18nextProvider } from 'react-i18next';
import i18n from '../__mocks__/i18nForTesting';



// Note: 'it' is an alias of test and is often used in examples
test('index test', done => {
  
  const component = renderer.create(
    <I18nextProvider i18n={ i18n }>
      <Header siteTitle="Aurora"/>
    </I18nextProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  i18n.changeLanguage("fr"); // not sure if this does anything going to have to check it out tuesday
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  done();

});