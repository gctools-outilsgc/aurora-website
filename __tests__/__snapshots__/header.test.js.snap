// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index test 1`] = `
<div
  className="fixed-top bg-white shadow-sm"
>
  <div>
    <div
      style={
        Object {
          "height": "50px",
          "padding": "10px",
        }
      }
    >
      <img
        alt="FIP"
        src="test-file-stub"
      />
      <div
        className="float-right"
      >
        <span
          className="float-left mr-3 d-none d-md-block"
        >
          <div
            className="search-form search-form-round"
            style={
              Object {
                "width": "300px",
              }
            }
          >
            <div
              className="dropdown"
              onKeyDown={[Function]}
            >
              <label
                className="sr-only"
                for="search"
              >
                Search
              </label>
              <input
                aria-expanded={false}
                aria-haspopup={true}
                className="form-control"
                id="search"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Search"
                type="text"
                value=""
              />
              <div
                aria-hidden={true}
                className="container-fluid dropdown-menu"
                role="menu"
                tabIndex="-1"
              >
                <li
                  className="list-group-item"
                  toggle={false}
                >
                  No results were found.
                </li>
              </div>
            </div>
          </div>
        </span>
        <span
          className="float-right"
        >
          <button
            className="languageSwitcherButton btn btn-secondary btn-md"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "550",
              }
            }
            type="button"
          >
            FR
          </button>
        </span>
      </div>
    </div>
  </div>
  <div>
    <nav
      className="navbar navbar-expand bg-white"
      style={
        Object {
          "paddingBottom": "0",
        }
      }
    >
      <Link
        className="text-dark"
        to="/#!"
      >
        <img
          alt="Logo"
          className="float-left"
          src="test-file-stub"
          style={
            Object {
              "margin": "5px",
              "width": "30px",
            }
          }
        />
        <span
          className="h2 d-none d-md-block float-left"
        >
           
          Aurora
           
        </span>
      </Link>
      <div
        className="navbar-collapse"
      >
        <ul
          className="mx-auto nav-site nav nav-tabs"
          style={
            Object {
              "borderBottom": "none",
            }
          }
        >
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/overview/whats-new"
            >
              Overview
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/#!"
            >
              Identity
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link active"
              to="/component/badges"
            >
              Component
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/content/content-guidelines"
            >
              Content
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/content/data/data-overview"
            >
              Data
            </Link>
          </li>
        </ul>
      </div>
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="d-none d-md-block nav-item"
        >
          <a
            className="nav-link"
            href="#!"
            onClick={[Function]}
          >
            GH
          </a>
        </li>
        <li
          className="d-none d-md-block nav-item"
        >
          <a
            className="nav-link"
            href="#!"
            onClick={[Function]}
          >
            DWNLD
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`index test 2`] = `
<div
  className="fixed-top bg-white shadow-sm"
>
  <div>
    <div
      style={
        Object {
          "height": "50px",
          "padding": "10px",
        }
      }
    >
      <img
        alt="FIP"
        src="test-file-stub"
      />
      <div
        className="float-right"
      >
        <span
          className="float-left mr-3 d-none d-md-block"
        >
          <div
            className="search-form search-form-round"
            style={
              Object {
                "width": "300px",
              }
            }
          >
            <div
              className="dropdown"
              onKeyDown={[Function]}
            >
              <label
                className="sr-only"
                for="search"
              >
                Chercher
              </label>
              <input
                aria-expanded={false}
                aria-haspopup={true}
                className="form-control"
                id="search"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Chercher"
                type="text"
                value=""
              />
              <div
                aria-hidden={true}
                className="container-fluid dropdown-menu"
                role="menu"
                tabIndex="-1"
              >
                <li
                  className="list-group-item"
                  toggle={false}
                >
                  No results were found.
                </li>
              </div>
            </div>
          </div>
        </span>
        <span
          className="float-right"
        >
          <button
            className="languageSwitcherButton btn btn-secondary btn-md"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "550",
              }
            }
            type="button"
          >
            FR
          </button>
        </span>
      </div>
    </div>
  </div>
  <div>
    <nav
      className="navbar navbar-expand bg-white"
      style={
        Object {
          "paddingBottom": "0",
        }
      }
    >
      <Link
        className="text-dark"
        to="/#!"
      >
        <img
          alt="Logo"
          className="float-left"
          src="test-file-stub"
          style={
            Object {
              "margin": "5px",
              "width": "30px",
            }
          }
        />
        <span
          className="h2 d-none d-md-block float-left"
        >
           
          Aurora
           
        </span>
      </Link>
      <div
        className="navbar-collapse"
      >
        <ul
          className="mx-auto nav-site nav nav-tabs"
          style={
            Object {
              "borderBottom": "none",
            }
          }
        >
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/overview/whats-new"
            >
              Overview
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/#!"
            >
              Identity
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link active"
              to="/component/badges"
            >
              Component
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/content/content-guidelines"
            >
              Content
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/content/data/data-overview"
            >
              Data
            </Link>
          </li>
        </ul>
      </div>
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="d-none d-md-block nav-item"
        >
          <a
            className="nav-link"
            href="#!"
            onClick={[Function]}
          >
            GH
          </a>
        </li>
        <li
          className="d-none d-md-block nav-item"
        >
          <a
            className="nav-link"
            href="#!"
            onClick={[Function]}
          >
            DWNLD
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
